File: aula_04/java/Log.java
import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.FileReader;
import java.io.FileWriter;
import java.io.IOException;

public class Log
{
    private String file;

    public Log(String file)
    {
        this.file = file;
    }

    public void writeLine(String payload) throws IOException
    {
        var writer = new BufferedWriter(
            new FileWriter(this.file, true)
        );

        writer.write(payload);
        writer.newLine();
        writer.close();
    }

    public String getLog() throws IOException
    {
        var reader = new BufferedReader(
            new FileReader(this.file)
        );

        String output = "";
        String line = "";
        while((line = reader.readLine()) != null)
        {
            output += line + "\n";
        }
        reader.close();

        return output;
    }
}

File: aula_04/python/Log.py
class Log:
    def __init__(self, file):
        self.__file = file

    def write_log(self, payload):
        file = open(self.__file, "a+")
        file.write(payload + "\n")
        file.close()

    def get_log(self):
        file = open(self.__file, "r")
        data = file.read()
        file.close()
        return data
